src/main/java/com/sequenceiq/cloudbreak/service/stack/StackFailureHandlerService.java

```diff
@@ -0,0 +1,133 @@
+                    }
+                    break;
+                default:
+                    throwError(stack, exceptions);
+                    break;
+            }
+        }
+    }
+
+    private void handleExceptions(Stack stack, List<ResourceRequestResult> exceptions) {
+        for (ResourceRequestResult exception : exceptions) {
+            InstanceGroup instanceGroup = instanceGroupRepository.findOne(exception.getInstanceGroup().getId());
+            instanceGroup.setNodeCount(instanceGroup.getNodeCount() - 1);
+            if (instanceGroup.getNodeCount() <= 0) {
+                LOGGER.info("InstanceGroup node count lower than 1 which is incorrect so error will thow");
+                throwError(stack, exceptions);
+                break;
+            } else {
+                LOGGER.info("InstanceGroup node count decreased with one and saved");
+                instanceGroupRepository.save(instanceGroup);
+                List<Resource> resourceList = new ArrayList<>();
+                for (Resource resource : exception.getResources()) {
+                    Resource newResource = resourceRepository.findByStackIdAndName(stack.getId(), resource.getResourceName(), resource.getResourceType());
+                    resourceList.add(newResource);
+                }
+                CloudPlatform cloudPlatform = stack.cloudPlatform();
+                ResourceBuilderInit resourceBuilderInit = resourceBuilderInits.get(cloudPlatform);
+                try {
+                    final DeleteContextObject dCO = resourceBuilderInit.deleteInit(stack);
+                    for (int i = instanceResourceBuilders.get(cloudPlatform).size() - 1; i >= 0; i--) {
+                        ResourceType resourceType = instanceResourceBuilders.get(cloudPlatform).get(i).resourceType();
+                        for (Resource tmpResource : resourceList) {
+                            if (resourceType.equals(tmpResource.getResourceType())) {
+                                instanceResourceBuilders.get(cloudPlatform).get(i).delete(tmpResource, dCO, stack.getRegion());
+                            }
+                        }
+                    }
+                } catch (Exception e) {
+                    MDCBuilder.buildMdcContext(stack);
+                    LOGGER.info("Resource can not be deleted: " + e.getMessage());
+                }
+                for (Resource tmpResource : resourceList) {
+                    resourceRepository.delete(tmpResource);
+                }
+            }
+        }
+    }
+
+    private void throwError(Stack stack, List<ResourceRequestResult> exceptions) {
+        Stack oneWithLists = stackRepository.findOneWithLists(stack.getId());
+        StringBuilder sb = new StringBuilder();
+        for (ResourceRequestResult resourceRequestResult : exceptions) {
+            sb.append(String.format("%s, ", resourceRequestResult.getException().orNull().getMessage()));
```

Comment:
Nope because of this :
https://github.com/sequenceiq/cloudbreak/blob/failure-handling-stack/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/ProvisionUtil.java#L48